### DISCUSSION TITLE
Why is the detection failing on some pictures?

### DISCUSSION THREAD

**Discussion by hakob-petro** [colour-science/colour-checker-detection#45] [Q&A] [answered]
Hi!

I apologize in advance for my English. I am trying to do color correction using the color checkers, as it described here: https://github.com/colour-science/colour-checker-detection/blob/develop/colour_checker_detection/examples/examples_detection.ipynb. 
So I have folder, where my pictures with color checkers in JPG format. And I just iterate through the list of pictures and try to detect the palette as it described in example. But when I'm calling the function `detect_colour_checkers_segmentation`, for some pictures it returns empty tuple. After I try to call function for that pictures separately, and sometimes it began to work correctly. 

I really can't figure out what this might be related to, so I'll be glad if someone explains to me what I'm doing wrong.

**Comment by KelSolaar** [comment, marked as answer]
Hi @hakob-petro,

The expectation for the current process to work is to have the Color Checker well centered and flat in the image. If it does not meet those requirements, e.g. it is not normal to the optical axis, or there is a lot of distortion, the detection will fail.

Cheers,

Thomas

**Reply by hakob-petro** [reply]
Yes, apparently this is the problem. 

Thanks a lot!
